File: SWGradientNodeObjC/SWGradientNode_iOS/ViewController.m
//
//  ViewController.m
//  SWGradientNode
//
//  Created by NSSimpleApps on 05.06.16.
//  Copyright © 2016 NSSimpleApps. All rights reserved.
//

#import "ViewController.h"
#import "SmoothGradientScene.h"
#import "SWGradientNode.h"


@interface ViewController ()

@property (weak, nonatomic) IBOutlet SKView *skView;

@end

@implementation ViewController

- (void)viewDidAppear:(BOOL)animated {
    
    [super viewDidAppear:animated];
        
    if (!self.skView.scene) {
        
        SmoothGradientScene *smoothGradientScene = [SmoothGradientScene sceneWithSize:self.skView.bounds.size];
        smoothGradientScene.scaleMode = SKSceneScaleModeAspectFill;//SKSceneScaleModeResizeFill;
        
        [self.skView presentScene:smoothGradientScene];
    }
}

- (IBAction)handlePanGesture:(UIPanGestureRecognizer *)sender {
    
    SKView *skView = (SKView *)sender.view;
    
    CGPoint point = [sender locationInView:skView];
    
    SKNode *skNode = [skView.scene nodeAtPoint:point];
    
    if ([skNode isKindOfClass:[SWGradientNode class]]) {
        
        SWGradientNode *node = (SWGradientNode *)skNode;
        node.center = CGPointMake(point.x/skView.frame.size.width, point.y/skView.frame.size.height);
    }
}

- (IBAction)handleRotationGesture:(UIRotationGestureRecognizer *)sender {
    
    SKView *skView = (SKView *)sender.view;
    
    CGPoint point = [sender locationInView:skView];
    
    SKNode *skNode = [skView.scene nodeAtPoint:point];
    
    if ([skNode isKindOfClass:[SWGradientNode class]]) {
        
        SWGradientNode *node = (SWGradientNode *)skNode;
        node.startAngle = sender.rotation;
    }
}

- (IBAction)sender:(UISlider *)sender {
    
    SmoothGradientScene *scene = (SmoothGradientScene *)self.skView.scene;
    
    CGPoint center = CGPointMake(0.5, 0.5);
    
    SWGradientNode *node = (SWGradientNode *)[scene nodeAtPoint:center];
    node.innerRadius = sender.value;
}

@end

File: SWGradientNodeSwift/SWGradientNode_iOS/ViewController.swift
//
//  ViewController.swift
//  SWGradientNode_iOS
//
//  Created by NSSimpleApps on 05.06.16.
//  Copyright © 2016 NSSimpleApps. All rights reserved.
//

import UIKit
import SpriteKit

class ViewController: UIViewController {
    
    @IBOutlet weak var skView: SKView!
    
    override func viewDidAppear(animated: Bool) {
        
        super.viewDidAppear(animated)
        
        let gradientScene = SmoothGradientScene()
        gradientScene.size = self.skView.bounds.size
        gradientScene.scaleMode = .AspectFill
        
        self.skView.presentScene(gradientScene)
    }
    
    @IBAction func handleRotationGesture(sender: UIRotationGestureRecognizer) {
        
        if let skView = sender.view as? SKView, let scene = skView.scene as? SmoothGradientScene {
            
            let location = sender.locationInView(skView)
            
            let node = scene.nodeAtPoint(location) as! SWGradientNode
            node.startAngle = Float(sender.rotation)
        }
    }
    
    @IBAction func handlePanGesture(sender: UIPanGestureRecognizer) {
        
        if let skView = sender.view as? SKView, let scene = skView.scene as? SmoothGradientScene {
            
            let location = sender.locationInView(skView)
            
            let center = CGPoint(x: location.x/skView.bounds.width, y: location.y/skView.bounds.height)
            
            let node = scene.nodeAtPoint(location) as! SWGradientNode
            node.center = center
        }
    }
    
    @IBAction func innerRadiusChanged(sender: UISlider) {
        
        if let scene = self.skView.scene as? SmoothGradientScene {
            
            let center = CGPoint(x: 0.5, y: 0.5)
            
            let node = scene.nodeAtPoint(center) as! SWGradientNode
            node.innerRadius = sender.value
        }
    }
}

